config/opal_load_platform.m4

```diff
@@ -97,7 +101,12 @@ AC_DEFUN([OPAL_LOAD_PLATFORM], [
             AC_SUBST(OPAL_PARAM_FROM_PLATFORM, "no")
         fi
 
-        patch_dir="${with_platform}.patches"
+        if test -d "$with_platform_patches_dir"; then
```

Comment:
I think you want to modify this test a bit.  If --with-platform-patches-dir=DIR is specified, and DIR does not exist, this should be an abort-configure kind of error (i.e., a human has asked for something that configure cannot deliver, therefore we should abort configure and left a human figure it out).